### PR TITLE
Reenable EsbuildCompilation on EXP_C

### PR DESCRIPTION
**Pull request by samouri**
**summary**
See https://github.com/ampproject/amphtml/pull/35816 for a PR that forces `esbuild` to be true and has all of CI green. Adds in one important fix for EXP_C, and then two fixes that only matter when it is the primary bundler:

1. ~(important): accept either unminified or minified name for `getAmpConfigForFile`. The distinction wasn't useful and forced us to pass in the minified name for minified builds (`srcFilename` is more ergonomic)~. Instead I pass `destFilename`
2. visual-diff:`test:true` check should also allow `!0` since that is the minified version of `true`
3. check-sourcemaps: esbuild has first line of the output on the first line instead of the second.

**confidence checks**
- prop mangling has been disabled for now. this is the riskiest optimization we could do, and my goal now is to create a baseline of how this branch performs _without_ it. If equivalent to the closure version, we can then pursue mangling separately.
- all automated tests pass
- manual testing of many sites (ping for details)
- [ ] full manual qa pass

### CODE REVIEW

**Review by jridgewell — APPROVED** [review]

File: build-system/tasks/check-sourcemaps.js [hunk 1]
```diff
@@ -115,7 +115,9 @@ function checkSourcemapMappings(sourcemapJson, map) {
 
   // Zeroth sub-array corresponds to ';' and has no mappings.
   // See https://www.npmjs.com/package/sourcemap-codec#usage
-  const firstLineMapping = decode(sourcemapJson.mappings)[1][0];
+  const firstLineMapping = decode(sourcemapJson.mappings)[
+    shouldUseClosure() ? 1 : 0
+  ][0];
```
Comment (rsimha) [inline build-system/tasks/check-sourcemaps.js:120]: Looks like the comment above is no longer accurate? Could we a) update it to specify how things are different with / without closure, and b) extract the first sub-array index to a separate variable for better readability?

File: build-system/tasks/prepend-global/index.js [hunk 2]
```diff
@@ -198,7 +198,7 @@ async function getConfig(
  * @return {!Promise<string>}
  */
 async function getAmpConfigForFile(filename, options) {
-  const targets = options.minify ? MINIFIED_TARGETS : UNMINIFIED_TARGETS;
+  const targets = MINIFIED_TARGETS.concat(UNMINIFIED_TARGETS);
```
Comment (rsimha) [inline build-system/tasks/prepend-global/index.js:201]: Could you explain why this change is necessary? (The target names are supposed to be mutually exclusive for minified vs. unminified.)

Comment (samouri) [reply build-system/tasks/prepend-global/index.js:201]: I don't need this update here. Can instead have `esbuildCompile` pass in the output name instead of src name.

IMO though, we can simplify the whole flow by not checking for `MINIFIED_TARGETS` at all, and only caring about UNMINIFIED

**Comment by samouri** [comment]
@rsimha: PTAL

File: build-system/tasks/check-sourcemaps.js [hunk 3]
```diff
@@ -113,9 +113,15 @@ function checkSourcemapMappings(sourcemapJson, map) {
...
-  const firstLineMapping = decode(sourcemapJson.mappings)[1][0];
+  // See https://www.npmjs.com/package/sourcemap-codec#usage.
+  const firstLineMapping = decode(sourcemapJson.mappings)[
+    // In closure builds there is a newline immediately after the AMP_CONFIG.
+    // This whole segment has no mapping to the original source.
+    // Therefore we must skip the zeroth sub-array, indicated by a ';'.
+
+    // In esbuild compiled binaries there is no newline after the config, so the first line is fair game.
+    shouldUseClosure() ? 1 : 0
+  ][0];
```
Comment (rsimha) [inline build-system/tasks/check-sourcemaps.js:124]: Suggestion for part b) of https://github.com/ampproject/amphtml/pull/36105#discussion_r713240645:
```suggestion
  // In closure builds there is a newline immediately after the AMP_CONFIG.
  // This whole segment has no mapping to the original source.
  // Therefore we must skip the zeroth sub-array, indicated by a ';'.
  // In esbuild compiled binaries there is no newline after the config, so the
  // first line is fair game.
  const firstLineIndex = shouldUseClosure() ? 1 : 0;

  // See https://www.npmjs.com/package/sourcemap-codec#usage.
  const firstLineMapping = decode(sourcemapJson.mappings)[firstLineIndex][0];

```

File: build-system/tasks/helpers.js [hunk 4]
```diff
@@ -417,7 +417,7 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
     };
   }
   const {banner, footer} = splitWrapper();
-  const config = await getAmpConfigForFile(srcFilename, options);
+  const config = await getAmpConfigForFile(destFilename, options);
```
Comment (rsimha) [inline build-system/tasks/helpers.js:420]: Thanks for fixing this.

**Review by rsimha — APPROVED** [review]
LGTM after a minor readability fix.

File: build-system/tasks/check-sourcemaps.js (hunk 3 above)
Comment (samouri) [reply build-system/tasks/check-sourcemaps.js:124]: Sure! I'll make this change

Comment (samouri) [reply build-system/tasks/check-sourcemaps.js]: Done!
